docker/docker_client.go

```diff
@@ -133,16 +133,34 @@ func dockerCertDir(ctx *types.SystemContext, hostPort string) string {
 		return ctx.DockerCertPath
 	}
 	var hostCertDir string
-	if ctx != nil && ctx.DockerPerHostCertDirPath != "" {
-		hostCertDir = ctx.DockerPerHostCertDirPath
-	} else if ctx != nil && ctx.RootForImplicitAbsolutePaths != "" {
-		hostCertDir = filepath.Join(ctx.RootForImplicitAbsolutePaths, systemPerHostCertDirPath)
-	} else {
-		hostCertDir = systemPerHostCertDirPath
+	for _, systemPerHostCertDirPath := range systemPerHostCertDirPaths {
+		if ctx != nil && ctx.DockerPerHostCertDirPath != "" {
```

Comment:
This case is independent of the loop variable, and should be moved out of the loop; if the caller specifies a path explicitly, we should use that one regardless of existence.

Reply:
fixed